src/helpers.c

```diff
@@ -164,16 +166,17 @@ int create_required_dirs(void)
 	}
 
 	if (missing) { // (re)create dirs
-		char *cmd;
+		char *param;
 
 		for (i = 0; i < STATE_DIR_COUNT; i++) {
-			string_or_die(&cmd, "mkdir -p %s/%s", state_dir, dirs[i]);
-			ret = system(cmd);
+			string_or_die(&param, "%s/%s", state_dir, dirs[i]);
+			char *const cmd[] = { "mkdir", "-p", param, NULL };
+			ret = system_argv(cmd);
```

Comment:
Feel free to take it: https://github.com/ikeydoherty/libnica/blob/master/src/files.c#L44